modin/backends/pandas/query_compiler.py

```diff
@@ -1154,12 +1150,35 @@ def unique(self):
     dt_days = MapFunction.register(_dt_prop_map("days"))
     dt_microseconds = MapFunction.register(_dt_prop_map("microseconds"))
     dt_nanoseconds = MapFunction.register(_dt_prop_map("nanoseconds"))
-    dt_components = MapFunction.register(_dt_prop_map("components"))
     dt_qyear = MapFunction.register(_dt_prop_map("qyear"))
     dt_start_time = MapFunction.register(_dt_prop_map("start_time"))
     dt_end_time = MapFunction.register(_dt_prop_map("end_time"))
     dt_to_timestamp = MapFunction.register(_dt_func_map("to_timestamp"))
 
+    def dt_tz(self):
+        new_modin_frame = self._modin_frame._apply_full_axis(
+            0,
+            lambda df: pandas.DataFrame([df.squeeze(axis=1).dt.tz]),
+            new_columns=self.columns,
+        )
+        return self.__constructor__(new_modin_frame)
+
+    def dt_freq(self):
+        new_modin_frame = self._modin_frame._apply_full_axis(
+            0,
+            lambda df: pandas.DataFrame([df.squeeze(axis=1).dt.freq]),
+            new_columns=self.columns,
+        )
+        return self.__constructor__(new_modin_frame)
```

Comment:
I think here we can apply `MapReduce` approach, it will apply a function to every partition and at the reduce phase we will get Series of scalars, for ex `["UTC +3", "UTC +3", "UTC +3", "UTC + 3"]` from which we can compute the scalar result. That will be faster than doing `apply_full_axis`, because we don't need to combine all partitions together and then splitting it again, which is decent work to do

Reply:
Thank you! This approach was applied.